tests/build_test.sh

```diff
@@ -53,8 +64,189 @@ tearDown()
   }
 }
 
+function test_parse_build_config_success_exit_code()
+{
+  build_config=()
+
+  parse_configuration "$KW_BUILD_CONFIG_SAMPLE" build_config
+  assertTrue 'kw failed to load a regular config file' "[ 0 -eq $? ]"
+}
+
+# Test if etc/kworkflow_template.config contains all the expected settings
+function test_parse_build_config_standard_config()
+{
+  build_config=()
+
+  declare -A expected_build_configurations=(
+    [arch]='x86_64'
+    [cpu_scaling_factor]='100'
+    [kernel_img_name]='bzImage'
+    [menu_config]='nconfig'
+    [doc_type]='htmldocs'
+  )
+
+  parse_configuration "$TMP_DIR/build.config" build_config
+  assertConfigurations build_config expected_build_configurations "$LINENO"
+
+  true # Reset return value
+}
+
+function assertConfigurations()
+{
+  declare -n configurations_ref=$1
+  declare -n expected_configurations_ref=$2
+  local lineno=${3:-LINENO}
+
+  # check if configurations is contained in expected_configurations
+  for k in "${!configurations_ref[@]}"; do
+    if [[ ${expected_configurations_ref[$k]+token} != token ]]; then
+      fail "($lineno): Did not expect setting '$k'."
+    elif [[ ${configurations_ref[$k]} != "${expected_configurations_ref[$k]}" ]]; then
+      fail "($lineno): Expected setting '${k}' to be '${expected_configurations_ref[$k]}' (found '${configurations_ref[$k]}')."
+    fi
+  done
+
+  # check if configurations has all expected_configurations keys
+  for k in "${!expected_configurations_ref[@]}"; do
+    if [[ ${configurations_ref[$k]+token} != token ]]; then
+      fail "($lineno): Expected setting '$k' to be present."
+    fi
+  done
+}
+
+# To test the order of config file loading, we will put a file named
+# kworkflow.config in each place, in order, and remove the previous one.
+# The order is: PWD, XDG_CONFIG_HOME, XDG_CONFIG_DIRS, KW_ETC_DIR
+function test_parse_configuration_files_loading_order()
+{
+  build_config=()
+
+  local expected
+  local original_dir="$PWD"
+
+  KW_ETC_DIR='1'
+  XDG_CONFIG_DIRS='2:3:4'
+  XDG_CONFIG_HOME='5'
```

Comment:
You probably want to restore the original values in the `setUp` function to avoid weird results in other tests.